src/test/java/io/vertx/core/dns/HostnameResolutionTest.java

```diff
@@ -834,6 +840,74 @@ private void testServerSelection(boolean rotateServers, boolean cache) throws Ex
     }
   }
 
+  private ResourceRecord createResourceRecord(String dnsName, String ipAddress) {
+    return new ResourceRecord() {
+
+      @Override
+      public String getDomainName() {
+        return dnsName;
+      }
+
+      @Override
+      public RecordType getRecordType() {
+        return RecordType.A;
+      }
+
+      @Override
+      public RecordClass getRecordClass() {
+        return RecordClass.IN;
+      }
+
+      @Override
+      public int getTimeToLive() {
+        return 100;
+      }
+
+      @Override
+      public String get(String s) {
+        return DnsAttribute.IP_ADDRESS.equals(s)? ipAddress : null;
+      }
+    };
+  }
+
+  @Test
+  public void testRoundRobinInetAddressSelection() throws Exception {
```

Comment:
I think we are missing a test where we check that setting setRoundRobinInetAddress to false only returns the first one.

Reply:
addressed, now we have 3 tests - default, on, and off.

Reply:
nice